module/os/operation_siren.py

```diff
@@ -116,6 +117,21 @@ def fleet_repair(self, revert=True):
         if revert and prev != self.zone:
             self.globe_goto(prev)
 
+    def handle_fleet_repair(self, enabled, threshold, revert=True):
+        if enabled:
+            self.hp_get()
+            check = [round(data, 2) <= threshold if use else False for data, use in zip(self.hp, self.hp_has_ship)]
+            if any(check):
+                logger.info('At least one ship is below threshold '
+                           f'{str(int(threshold * 100))}%, retreating '
+                            'to nearest azur port for repairs')
+                self.fleet_repair(revert=revert)
+                self.hp_reset()
+            else:
+                logger.info('No ship found to be below threshold '
+                           f'{str(int(threshold * 100))}%, continue '
+                            'OS exploration')
+
```

Comment:
Use the setting directly in this method.
```python
def handle_fleet_repair(self):
    if self.config.ENABLE_OS_SETTING_REPAIR:
        self.hp_get()
        check = [round(data, 2) <= self.config.OS_SETTING_REPAIR_THRESHOLD if use else False for data, use in zip(self.hp, self.hp_has_ship)]
        pass
```

Reply:
You can even delete ENABLE_OS_SETTING_REPAIR, fleet repair will be enabled when OS_SETTING_REPAIR_THRESHOLD > 0
```python
def handle_fleet_repair(self):
    if self.config.OS_SETTING_REPAIR_THRESHOLD > 0:
        self.hp_get()
        check = [round(data, 2) <= self.config.OS_SETTING_REPAIR_THRESHOLD if use else False for data, use in zip(self.hp, self.hp_has_ship)]
        pass
```

Reply:
This is true, just using threshold should be sufficient and play the role as both enabler and configurable.

Thanks